### PR TITLE
extract package from the plist file

### PR DESCRIPTION
**Pull request by SergiX44**
Add a way to automatically extract the package name from the plist file while uploading a new build.
For Android is not implemented yet as it's not needed, and outside the scope of the bug.

### CODE REVIEW

File: app/Platforms/AndroidPlatform.php [hunk 1]
```diff
@@ -28,4 +29,14 @@ public function getDownloadUrl(Application $application, Build $build, Cloud $st
...
         return $this->getFileUrl($build, $storage);
     }
+
+    /**
+     * @param  UploadedFile  $file
+     * @return string|null
+     */
+    public function getPackage(UploadedFile $file): ?string
+    {
+        // TODO: Implement getPackage() method.
```
Comment (sinisimattia) [inline app/Platforms/AndroidPlatform.php:39]: I know it's obvious but we should create a tech debt story for this

**Review by sinisimattia — APPROVED** [review]
Really clean and looks good! :+1:

File: app/Platforms/IOSPlatform.php [hunk 2]
```diff
@@ -30,4 +35,32 @@ public function getDownloadUrl(Application $application, Build $build, Cloud $st
...
         return 'itms-services://?action=download-manifest&url='.urlencode($plistUrl);
     }
+
+    /**
+     * @param  UploadedFile  $file
+     * @return string|null
+     */
+    public function getPackage(UploadedFile $file): ?string
+    {
+        $archive = new ZipArchive();
```
Comment (IgorErmakov) [inline app/Platforms/IOSPlatform.php:45]: Just curious, how big could be the file? is ok to unpack it in ram?

File: app/Platforms/IOSPlatform.php [hunk 3]
```diff
@@ -30,4 +35,32 @@ public function getDownloadUrl(Application $application, Build $build, Cloud $st
...
+        $archive->open($file->getRealPath());
+        $content = null;
+
+        for ($i = 0; $i < $archive->numFiles; $i++) {
+            $entry = $archive->statIndex($i);
+            if (fnmatch('*/*.app/Info.plist', $entry['name'])) {
```
Comment (IgorErmakov) [inline app/Platforms/IOSPlatform.php:51]: that's awesome!

**Review by IgorErmakov — APPROVED** [review]
Looks good !

File: app/Platforms/IOSPlatform.php (hunk 2 above)
Comment (SergiX44) [reply app/Platforms/IOSPlatform.php:45]: The info.plist is just a couple of KBs

**Pull request merged**